src/openrct2/ride/RideRatings.cpp

```diff
@@ -762,6 +767,36 @@ static void ride_ratings_calculate(Ride* ride)
         ride->ratings.nausea = max(0, ride->ratings.nausea);
     }
 #endif
+
+#ifdef ENABLE_SCRIPTING
+    auto& hookEngine = GetContext()->GetScriptEngine().GetHookEngine();
+    if (hookEngine.HasSubscriptions(HOOK_TYPE::RIDE_RATINGS_CALCULATE))
+    {
+        auto ctx = GetContext()->GetScriptEngine().GetContext();
+        auto originalExcitement = ride->excitement;
+        auto originalIntensity = ride->intensity;
+        auto originalNausea = ride->nausea;
+
+        // Create event args object
+        auto obj = DukObject(ctx);
+        obj.Set("rideId", ride->id);
+        obj.Set("excitement", originalExcitement);
+        obj.Set("intensity", originalIntensity);
+        obj.Set("nausea", originalNausea);
+
+        // Call the subscriptions
+        auto e = obj.Take();
+        hookEngine.Call(HOOK_TYPE::RIDE_RATINGS_CALCULATE, e, true);
+
+        auto scriptExcitement = AsOrDefault(e["excitement"], static_cast<int32_t>(originalExcitement));
+        auto scriptIntensity = AsOrDefault(e["intensity"], static_cast<int32_t>(originalIntensity));
+        auto scriptNausea = AsOrDefault(e["nausea"], static_cast<int32_t>(originalNausea));
+
+        ride->excitement = std::clamp(scriptExcitement, 0, INT16_MAX);
+        ride->intensity = std::clamp(scriptIntensity, 0, INT16_MAX);
+        ride->nausea = std::clamp(scriptNausea, 0, INT16_MAX);
```

Comment:
You need to explicitly say what the type is for these `std::clamp` as `INT16_MAX` is a different type to the literal 0.